### ISSUE TITLE
Implement software profiles for mice

### ISSUE THREAD

**Issue by whot** [libratbag/libratbag#475]
Brought up in https://github.com/libratbag/libratbag/issues/460#issuecomment-378757968, this is the tracker bug for software profiles in libratbag.

This was originally frowned upon, we said only devices with hardware-stored profiles. Now that we are a daemon instead of a library, software profiles should be doable. Note that the device configuration is still stored/handled in the device, it just doesn't survive an unplug. So all we need to do is store the configuration on the host and then load it into the device on plug. Otherwise, things stay the same, this is *not* about re-routing events in ratbagd.

The two main components are:
- [ ] a backend that allows to read/write device configuration, ideally all devices use the same storage format and queries
- [ ] a plan on how to trigger ratbagd and thus configuration writing on device hotplug.

**Comment by FFY00** [comment]
Yes, this would also be really helpful for devices that don't support reading the device settings (looking at you steelseries). I was actually able to read the DPI values in my Steelseries Rival 310 but not anything else.
